benchmark/case/curry.expect

```diff
@@ -0,0 +1,56 @@
+// Test deeply nested lambdas.
+main() {
+  success(
+        (x1) =>
+            (x2) =>
+                (x3) =>
+                    (x4) =>
+                        (x5) =>
+                            (x6) =>
+                                (x7) =>
+                                    (
+                                      x8,
+                                    ) =>
```

Comment:
What is happening here?

Reply:
Oops! The short answer is that in the benchmark case file, I accidentally copy/pasted:

```
40 columns                              |
```

Which is trying to fit this into 40 columns, unlike the original regression test which is a full 80 columns. That's where there's a weird split here. I fixed that.

The longer answer is that it seems like splitting multiple of the parameter lists would help it pack things into the page width in this weird case, but the solver doesn't seem to figure that out. I believe that's at least in part because the AssignPiece used for `=>` doesn't have the parameter list as the left operand of the AssignPiece itself and instead makes that part of the surrounding FunctionPiece.

I spent a little time poking around at that this morning but didn't make anything better. I think it probably makes sense to worry about that in the context of #1466.

Reply:
It's probably not a big deal if we have a weird behavior right at the edge of the intended width. It does stand out, though, seeing the 1 level in the middle of the nesting formatted differently from the identical syntax before and after it.